### PR TITLE
Custom Column Types

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: painless_sqlalchemy/columns/AreaType.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        # check if clockwise
+        direction = 0
+        for i in range(len(area) - 1):
+            direction += (
+                area[i + 1][0] - area[i][0]
+            ) * (
+                area[i + 1][1] + area[i][1]
+            )
+        # reverse if counter clockwise
+        return area[::-1] if direction < 0 else area
```
Comment (MacMcIrish) [inline painless_sqlalchemy/columns/AreaType.py]: Should we raise an error here? Could be considered "magic"

Comment (MacMcIrish) [reply painless_sqlalchemy/columns/AreaType.py]: Thinking about naming here. We could have a flag `input_clockwise: bool` which would reverse the direction, but `enforce_clockwise` to me communicates "This is to make sure the input data is clockwise"

File: painless_sqlalchemy/util/LocationUtil.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    if not isinstance(latlong, (tuple, list)) or len(latlong) != 2:
+        raise ValueError("Invalid latlong format.")
+    return validate_latitude(latlong[0]), validate_longitude(latlong[1])
+
+
+def haversine(lat1, lon1, lat2, lon2):
+    """
+    Calculate the great circle distance between two points
+    on the earth (specified in decimal degrees).
+    Returns distance in km.
```
Comment (MacMcIrish) [inline painless_sqlalchemy/util/LocationUtil.py]: Should we modify this to return in `m` instead? From previous in person discussion about keeping `m` as distance standard

File: painless_sqlalchemy/columns/AreaType.py (hunk 1 above)
Comment (simlu) [reply painless_sqlalchemy/columns/AreaType.py]: Changed it to a boolean flag

File: painless_sqlalchemy/columns/AreaType.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    @classmethod
+    def as_postgis(cls, area):
+        area = cls.rec_round(area)  # round elements
+        area = cls.enforce_clockwise(area)  # enforce orientation
```
[the remark this answers is not in the document]
Comment (simlu) [reply painless_sqlalchemy/columns/AreaType.py]: This was changed now, so no longer possible. But agreed rgd comments

File: painless_sqlalchemy/util/LocationUtil.py (hunk 2 above)
Comment (simlu) [reply painless_sqlalchemy/util/LocationUtil.py]: done

**Pull request merged**